lib/spack/spack/relocate.py

```diff
@@ -289,7 +289,7 @@ def relocate_text(path_names, old_dir, new_dir):
     """
     Replace old path with new path in text file path_name
     """
-    filter_file("r'%s'" % old_dir, "r'%s'" % new_dir,
```

Comment:
What was this originally intended to do? Did it break for certain inputs? If so, could you mention an example? To be clear I'm just looking for an answer here in the discussion thread (vs. updating the commit).

Reply:
The text is not getting changed during tarball extraction.

Reply:
Would you be able to add a unit test to confirm this function changes text as expected?

Reply:
Moved to https://github.com/spack/spack/pull/6510